### PR TITLE
Update hubert doc

### PR DESCRIPTION
**Pull request by NewGamezzz**
## What did you change?

- Added a detailed breakdown of the HuBERT recipe flow, including feature dumping and K-means clustering logic.
- Documented the DiceHubert distillation method, explaining how to use pretrained models to bypass early iterations.
- Added an Evaluation section specifically for the SUPERB benchmark using the S3PRL upstream espnet_hubert_local.

---

## Why did you make this change?

The current HuBERT documentation does not provide the details of each stage, making it hard to understand the script. These updates provide clear instructions for researchers looking to replicate the HuBERT and DiceHubert results, specifically within the ESPnet2 framework.

---

## Is your PR small enough?

Yes.

---

### CODE REVIEW

**Comment by sw005320** [comment]
Thanks, @NewGamezzz!

@wanchichen, can you review this PR?

**Comment by Fhrozen** [comment]
@claude review

**Comment by sw005320** [comment]
@wanchichen, this is a reminder

**Review by sw005320 — COMMENTED** [review]
Thanks @NewGamezzz — this is a real improvement to the template. I checked the documented flow against the scripts and most of it holds up: the stage 1-9 numbering matches `hubert.sh`, the five sub-steps match `perform_kmeans.sh` (dump / learn k-means / generate labels / measure quality / prepare dictionary), and the output locations you name (`dump/hubert_feats/`, the k-means model under `exp/`, the dictionary under `data/`) all match what the scripts actually write. The `--download_model` shortcut for skipping iteration 1 is used exactly where you say it is. On the evaluation side, `espnet_hubert_local` does exist as an s3prl upstream, and `-k` / `-g` are the right flags for the checkpoint and config. CI is green.

Two option names are wrong, though, and both would send a user down a dead end.

### 1. `--feat_type` does not exist

> The HuBERT recipe supports only `--feat_type raw`.

The option is `--feats_type` (`hubert.sh:55`). With `--feat_type` the user gets an "unknown option" error from `parse_options.sh`.

### 2. `n_clusters_lists` does not exist

> You can set the number of clusters using `n_clusters_lists`

The CLI option is `--n_clusters` (`hubert.sh:74`, documented in the help text at `hubert.sh:138`). `n_clusters_list` — singular, no leading dashes — is an internal bash array built at `hubert.sh:195` and is not settable from the command line. Since the surrounding sentence tells the user to follow the same space-separated format as `--features_km`, it is worth writing it as `--n_clusters "100 500 500"` to match the help text.

### Minor

- **"K-mean" should be "K-means"** in the two sub-section headings ("Train K-mean Clustering", "Generate K-mean pseudo-labels"). `perform_kmeans.sh` itself logs "K-means". The table-of-contents anchors need the same edit.
- **The quality-evaluation step is skipped entirely without `--alignment_phoneme_dir`.** "If phoneme labels are available" is accurate, but naming the option would save the reader a trip into `perform_kmeans.sh:300`.
- **`use_amp: false` in the new distillation config.** The base `train_ssl_torchaudiohubert_base_960h_pretrain_it1.yaml` uses `use_amp: true`; everything else in the new file is either the smaller student geometry or environment tuning. Is disabling AMP deliberate, or carried over from a local debug run? It roughly doubles training time if unintentional.
- `[DiceHuBERT](https://arxiv.org/pdf/2507.02911)` would be better as the `/abs/` URL, and the sentence beginning "Setting --train_start_iter 2" leaves the flags outside a code span while the rest of the document consistently uses one.

Nice catch on the `hubert11` → `hubert1` typo in the opening line, by the way.

**Comment by NewGamezzz** [comment]
Thanks @sw005320. Both wrong options are fixed (`--feats_type`, and `--n_clusters "100 500 500"`
to match the help text), and I applied all the minor suggestions: "K-means" in the headings and
table of content, `--alignment_phoneme_dir` named in the quality step, the `/abs/` link, and the
flags in a code span.

I would like to note that `use_amp: false` is intentional. I previously encountered training instability with
`use_amp: true` on this setup, so I disabled it.

Here is the training loss with `use_amp: true`
image

and here is the training loss with `use_amp: false`
image

File: egs2/TEMPLATE/hubert1/README.md [hunk 1]
```diff
@@ -1,6 +1,113 @@
...
+```sh
+./run.sh --stage 1 --stop-stage 4
+```
+Following the original HuBERT setup, the student model is trained using the same procedure as iteration 2, with a smaller model size. To generate pseudo-labels without training HuBERT from scratch, you can download a pretrained model as follows:
+```sh
+./run.sh --stage 5 --stop-stage 5 --train_start_iter 2 --train_stop_iter 2 --download_model simpleoier/simpleoier_librispeech_hubert_iter1_train_ssl_torchaudiohubert_base_960h_pretrain_it1_raw
+```
+Setting `--train_start_iter 2 --train_stop_iter 2` ensures that only iteration 2 is executed. After generating pseudo-labels, continue with the standard training steps:
+```sh
+./run.sh --stage 6 --stop-stage 7 --train_start_iter 2 --train_stop_iter 2
```
[the remark this answers is not in the document]
Comment (NewGamezzz) [reply egs2/TEMPLATE/hubert1/README.md]: This is correct. I have fixed the readme to explicitly pass the DiceHuBert's config.

File: egs2/TEMPLATE/hubert1/README.md [hunk 2]
```diff
@@ -1,6 +1,113 @@
...
+This recipe does not include a built-in evaluation stage. However, the trained HuBERT model is compatible with the SUPERB benchmark.
+
+First, clone the [s3prl repository](https://github.com/s3prl/s3prl/tree/main). Since `s3prl` supports HuBERT models, you can set the upstream model to `espnet_hubert_local` as follows:
+```sh
+python s3prl/run_downstream.py \
+    -m train \
+    -d $TASK \
+    -u espnet_hubert_local \
+    -k "$CKPT" \
+    -g "$CONFIG"
```
[the remark this answers is not in the document]
Comment (NewGamezzz) [reply egs2/TEMPLATE/hubert1/README.md]: You are right, I confirmed this by running it. Without `-n` the output does go to `result/downstream/None`, and `-c` resolves against the working directory rather than the cloned S3PRL config. Fixed: the command now runs from the `s3prl/s3prl` package directory and passes `-n "${TASK}_espnet_hubert"` and `-c "downstream/$TASK/config.yaml"` explicitly, so each task writes to its own directory. `-g` is unchanged, it is the upstream model config as you note.

**Comment by sw005320** [comment]
> I would like to note that use_amp: false is intentional. I previously encountered training instability with
use_amp: true on this setup, so I disabled it.

Cool. Can you add it to the comment in READM.md?

**Comment by NewGamezzz** [comment]
Done! I noted this in the README.md and also added a comment at the top of the distillation config file.

**Review by sw005320 — APPROVED** [review]
Thanks, @NewGamezzz!
This will be merged after the CI check
